compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -1215,10 +1220,107 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         );
 
         self.suggest_using_local_if_applicable(&mut err, location, issued_borrow, explanation);
+        self.suggest_copy_for_type_in_cloned_ref(&mut err, place);
 
         err
     }
 
+    fn suggest_copy_for_type_in_cloned_ref(&self, err: &mut Diag<'tcx>, place: Place<'tcx>) {
+        let tcx = self.infcx.tcx;
+        let hir = tcx.hir();
+        if let Some(body_id) = tcx.hir_node(self.mir_hir_id()).body_id() {
```

Comment:
```suggestion
        let Some(body_id) = tcx.hir_node(self.mir_hir_id()).body_id() else { return };
```

just for making the function easier to read top to bottom (so I don't have to remember we're in a scope)